src/firmware/app/control/trajectory_planner_test.cpp

```diff
@@ -1288,50 +768,1631 @@ TEST_F(TrajectoryPlannerTest, velocity_trajectory_curved_path_low_acceleration)
+    VelocityTrajectoryElement_t
+        velocity_trajectory_elements[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+
+    position_trajectory.trajectory_elements = position_trajectory_elements;
+    velocity_trajectory.trajectory_elements = velocity_trajectory_elements;
+
+    position_trajectory.path_parameters = path_parameters;
+    velocity_trajectory.path_parameters = path_parameters;
+
+    app_trajectory_planner_getMaxAllowableSpeedProfile(
+        position_trajectory.path_parameters.path,
+        position_trajectory.path_parameters.num_segments,
+        position_trajectory.path_parameters.t_start,
+        position_trajectory.path_parameters.t_end,
+        position_trajectory.path_parameters.max_allowable_linear_acceleration,
+        max_allowable_speed_profile);
+    app_trajectory_planner_generateStatesAndReturnSegmentLengths(&position_trajectory,
+                                                                 segment_lengths);
+
+    app_trajectory_planner_createForwardsContinuousLinearSpeedProfile(
+        &position_trajectory, max_allowable_speed_profile, segment_lengths);
+    app_trajectory_planner_createForwardsContinuousAngularSpeedProfile(
+        &position_trajectory, segment_lengths);
+
+    app_trajectory_planner_modifyTrajectoryToBackwardsContinuous(&position_trajectory,
+                                                                 segment_lengths);
+    app_trajectory_planner_generatePositionTrajectoryTimeProfile(&position_trajectory,
+                                                                 segment_lengths);
+
+    app_trajectory_planner_generateVelocityTrajectory(&position_trajectory,
+                                                      &velocity_trajectory);
+
+    std::vector<Vector> directions =
+        getDirectionVectorsFromPositionTrajectory(&position_trajectory);
+    const float segment_length = 9 * sqrt(2) / (path_parameters.num_segments - 1);
+
+    for (unsigned int i = 0; i < velocity_trajectory.path_parameters.num_segments; i++)
+    {
+        EXPECT_FLOAT_EQ(velocity_trajectory.trajectory_elements[i].angular_velocity, 0);
+        EXPECT_FLOAT_EQ(velocity_trajectory.trajectory_elements[i].linear_velocity.x,
+                        directions[i].x());
+        EXPECT_FLOAT_EQ(velocity_trajectory.trajectory_elements[i].linear_velocity.y,
+                        directions[i].y());
+        EXPECT_NEAR(velocity_trajectory.trajectory_elements[i].time, i * segment_length,
+                    0.00001);
+    }
+}
+
+TEST_F(TrajectoryPlannerTest,
+       test_generate_velocity_trajectory_no_path_linear_orientation)
+{
+    // Create a scenario where the linear path parameters will limit the speed of the
+    // trajectory by using angular parameters that are much greater
+    FirmwareRobotPathParameters_t path_parameters;
+    path_parameters.max_allowable_angular_acceleration =
+        100;  // Make linear acceleration large to reach steady state fast
```

Comment:
Similar comment here about end-of-line comments, this is a good example of how formatting can knacker 'em

Reply:
Outdated.